Wrappers/Python/ccpi/web_viewer/trame_viewer.py

```diff
@@ -0,0 +1,772 @@
+
+        # Reset all the buttons and camera
+        self.reset_defaults()
+
+    def load_image(self, image_file: str):
+        reader = vtkMetaImageReader()
+        reader.SetFileName(image_file)
+        reader.Update()
+        self.image = reader.GetOutput()
+        self.cil_viewer.setInput3DData(self.image)
+
+    def load_nexus_file(self, file_name):
+        reader = cilHDF5ResampleReader()
+        reader.SetFileName(file_name)
+        reader.SetDatasetName('entry1/tomo_entry/data/data')
+        reader.SetTargetSize(256 * 256 * 256)
+        reader.Update()
+        self.image = reader.GetOutput()
+        self.cil_viewer.setInput3DData(self.image)
+
+    def switch_render(self):
+        if not self.cil_viewer.volume_render_initialised:
+            self.cil_viewer.installVolumeRenderActorPipeline()
+
+        if self.cil_viewer.volume.GetVisibility():
+            self.cil_viewer.volume.VisibilityOff()
+            self.cil_viewer.light.SwitchOff()
+        else:
+            self.cil_viewer.volume.VisibilityOn()
+            self.cil_viewer.light.SwitchOn()
+        self.cil_viewer.updatePipeline()
+        self.html_view.update()
+
+    def switch_slice(self):
+        if self.cil_viewer.imageSlice.GetVisibility():
+            self.cil_viewer.imageSlice.VisibilityOff()
+        else:
+            self.cil_viewer.imageSlice.VisibilityOn()
+        self.cil_viewer.updatePipeline()
+        self.html_view.update()
+
+    def switch_to_orientation(self, slice_orientation):
+        self.cil_viewer.sliceOrientation = slice_orientation
+        self.cil_viewer.updatePipeline()
+        self.html_view.update()
+
+    def change_colour_map(self, cmap):
+        self.cil_viewer.setVolumeColorMapName(cmap)
+        self.cil_viewer.updatePipeline()
+        self.html_view.update()
+
+    def set_opacity_mapping(self, opacity):
+        self.update_windowing_defaults(opacity)
+        self.cil_viewer.setVolumeRenderOpacityMethod(opacity)
+        self.html_view.update()
+
+    def change_windowing(self, min_value, max_value, windowing_method="scalar"):
+        if self.windowing_slider_is_percentage:
+            if windowing_method == "scalar":
+                self.cil_viewer.setScalarOpacityPercentiles(min_value, max_value)
+            else:
+                self.cil_viewer.setGradientOpacityPercentiles(min_value, max_value)
+        else:
+            if windowing_method == "scalar":
+                self.cil_viewer.setScalarOpacityWindow(min_value, max_value)
+            else:
+                self.cil_viewer.setGradientOpacityWindow(min_value, max_value)
+        if hasattr(self, "html_view"):
+            self.html_view.update()
+
+    def reset_cam(self):
+        self.cil_viewer.adjustCamera(resetcamera=True)
+        if hasattr(self, "original_cam_data"):
+            self.original_cam_data.copy_data_to_other_camera(self.cil_viewer.ren.GetActiveCamera())
```

Comment:
We may be able to use:
https://github.com/vais-ral/CILViewer/blob/6ac6efa69bd1e15ce0ef8cbca77fe6e5c572fdf5/Wrappers/Python/ccpi/viewer/CILViewer.py#L738

instead of the need for CamData.

This is something I added to the viewer recently. When the pipeline is first set up, a copy of the camera is made in the 3D viewer ( https://github.com/vais-ral/CILViewer/blob/6ac6efa69bd1e15ce0ef8cbca77fe6e5c572fdf5/Wrappers/Python/ccpi/viewer/CILViewer.py#L731) , then https://github.com/vais-ral/CILViewer/blob/6ac6efa69bd1e15ce0ef8cbca77fe6e5c572fdf5/Wrappers/Python/ccpi/viewer/CILViewer.py#L738 can be used to set the camera to use that default.

Reply:
This doesn't seem to solve the same problem as reset_cam resets the position to the original position not just the most recent pipeline invocation as that is used for some other features in the web_app.